azurelinuxagent/ga/exthandlers.py

```diff
@@ -958,42 +957,52 @@ def report_ext_handlers_status(self, incarnation_changed=False, vm_agent_update_
                       message=msg)
             return None
 
-    def write_ext_handlers_status_to_info_file(self, vm_status):
-        status_path = os.path.join(conf.get_lib_dir(), AGENT_STATUS_FILE.format(self.protocol.get_incarnation()))
-        status_blob_data = self.protocol.get_status_blob_data()
-        data = dict()
-        if status_blob_data is not None:
-            data = json.loads(status_blob_data)
+    def write_ext_handlers_status_to_info_file(self, vm_status, incarnation_changed):
+        status_file = os.path.join(conf.get_lib_dir(), AGENT_STATUS_FILE)
 
-        # Populating the fields that does not come from vm_status or status_blob_data
-        _metadataNotSentToCRP = {
-            "agentName": AGENT_NAME,
-            "daemonVersion": str(version.get_daemon_version()),
-            "pythonVersion": "Python: {0}.{1}.{2}".format(PY_VERSION_MAJOR, PY_VERSION_MINOR, PY_VERSION_MICRO),
-            "extensionSupportedFeatures": [name for name, _ in
-                                         get_agent_supported_features_list_for_extensions().items()]
-        }
-        data["_metadataNotSentToCRP"] = _metadataNotSentToCRP
+        if os.path.exists(status_file) and incarnation_changed:
+            # On new goal state, move the last status report for the previous goal state to the history folder
+            last_modified = os.path.getmtime(status_file)
+            timestamp = datetime.datetime.utcfromtimestamp(last_modified).isoformat()
+            GoalStateHistory(timestamp).save_status(status_file)
+
+        # Now create/overwrite the status file; this file is kept for debugging purposes only
+        status_blob_text = self.protocol.get_status_blob_data()
+        if status_blob_text is None:
+            status_blob_text = ""
+
+        debug_info = ExtHandlersHandler._get_status_debug_info(vm_status)
```

Comment:
while working on the history i noticed that we were modifying the status we report to crp before saving it to the status file.

i changed this to preserve the status exactly as we save it to the blob